subprojects/core/src/integTest/groovy/org/gradle/invocation/BuildStartedDeprecatedIntegrationTest.groovy

```diff
@@ -0,0 +1,147 @@
+/*
+ * Copyright 2019 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.invocation
+
+import org.gradle.integtests.fixtures.AbstractIntegrationSpec
+import spock.lang.Unroll
+
+class BuildStartedDeprecatedIntegrationTest extends AbstractIntegrationSpec {
+    private static final String INIT_FILE_NAME = "init.gradle"
+
+    @Unroll
+    def "shows deprecation warning when adding build listener through Gradle.addBuildListener that override the BuildAdapter#buildStarted after the build was started (#fromScript)"() {
+        def initScriptFile = file(INIT_FILE_NAME).touch()
+        file(scriptFile) << """
+            gradle.addBuildListener(new BuildAdapter() {
+                void buildStarted(Gradle gradle) {
+                    assert false
+                }
+            })
+        """
+
+        executer.usingInitScript(initScriptFile)
+
+        expect:
+        executer.expectDeprecationWarning()
```

Comment:
Can you please use the new API and expect the deprecation warning with a message?

Reply:
Done in https://github.com/gradle/gradle/commit/7f31c7126f589383495ce0491118d6be83857287